### PR TITLE
Json changes for the new temperature system

### PR DESCRIPTION
**Pull request by Hirmuolio**
#### Summary

 ```SUMMARY: Infrastructure "Adds "primary_material" to foods that need it."```

#### Purpose of change
The new temperature system usually gets the item thermal attributes from its materials.
But the materials do not always reflect the properties of the item.

#### Describe the solution
This adds "primary_material" to these items. The properties of this material are used for temperature calculations of this item.

New materials were added. These are currently not used as "normal" material for any item. But they *should* work for that too.

New materials:

* Dried vegetables - All dried plant/mushroom food. Vitamins/fuel copied from fruit.
* Cured meat - Cured/dried meat products. Vitamins/fuel copied from meat.
* Candy - Candy. Burn data copied from junk food.
* Processed food - Complex cooked foods like pies and such. Basically for foods  that would get odd values without this (such as having vegetable/milk in materials) Burn data copied from junk food.
* Cheese - Cheese. Vitamins/burn copied from milk.
* Ice cream - Ice cream. Just to make all ice creams identical. Burn copied from milk.

#### Describe alternatives you've considered
Change the materials of the items so that the average of all of its materials reflects the properties of the item.

#### Additional context
Won't do anything without #27387

Some items that are very similar may seem to have different materials but the result is that they are very similar. For example flesh+wheat is almost same as processed food. Unless the used materials are completely wrong the result is close enough.

I may have missed some items. I didn't do much testing beyond seeing that the game still loads with this change and that the "primary_material" system works.

### CODE REVIEW

File: data/json/materials.json [hunk 1]
```diff
@@ -1062,6 +1125,125 @@
...
+    "bash_resist": 0,
+    "cut_resist": 0,
+    "acid_resist": 0,
+    "fire_resist": 0,
+    "elec_resist": 0,
+    "chip_resist": 0,
+    "dmg_adj": [ "lightly damaged", "damaged", "very damaged", "thoroughly damaged" ],
+    "bash_dmg_verb": "smashed",
+    "cut_dmg_verb": "cut",
+    "vitamins": [ [ "vitB", 0.5 ], [ "calcium", 1.5 ], [ "iron", 0.1 ] ],
```
Comment (KorGgenT) [inline data/json/materials.json]: these can only be ints

Comment (Hirmuolio) [reply data/json/materials.json]: specific_heat_liquid and specific_heat_solid? They seem to be working fine in their current form.

**Comment by KongMD-Steam** [comment]
I think the 'hydrocarbons' material needs the tags below applied. After the temperature PR was merged, I saw screenshots with a user frozen gasoline. I suspect it was causing a crash they reported in Discord.
```
specific_heat_solid
specific_heat_liquid
```

**Comment by Hirmuolio** [comment]
Game seems to crash when adding gasoline to a container. I'll look into that.

I didn't get any frozen gasoline. Maybe in some specific process it gets its temperature set even though only food items should run through those processes.

Adding the values in json won't help with that. Those only change how fast the heating/cooling happens and that does not happen at all for gasoline.

File: data/json/materials.json (hunk 1 above)
Comment (kevingranade) [reply data/json/materials.json]: Pretty sure he means vitamins.

Comment (Hirmuolio) [reply data/json/materials.json]: The vitamin values were copy-pasted from already existing materials. Cheese for example from dairy.

Comment (KorGgenT) [reply data/json/materials.json]: oh my apologies, with the large nutrition overhaul i forgot that materials themselves can have vitamins. maybe we should think about if that's a necessary thing anymore... but that's not really a thing for this PR

**Pull request merged**